### DISCUSSION TITLE
Migrating from Sonoff Dongle-E to Sonoff Dongle-P

### DISCUSSION THREAD

**Discussion by yandy-r** [Koenkk/zigbee2mqtt#17370] [Devices]
Hi,

So I know, based on the documentation the backup/restore feature is not available going from EZSP to ZSTACK, and I will need to repair everything. That's not that big of an issue for me, the biggest issue are the automations that depend on those devices. I'm currently using it with Homekit via the homebridge-z2m plugin.

My question(s) are;

- Will re-pairing the devices to z2m change the device ids? Or will they stay the same so that once I bring homebridge backup, it'll recognize the same devices?

In other words.

- I shut down the homebridge container for as long as it takes me to repair all the devices.
- I name them the same (though probably names shouldn't matter)
- I bring homebridge backup, will the devices be recognized as the same or new homekit accessories and me having to re-do all the automations or at least finding them and adding the accessories back in?

Has anyone attempted this? Is it worth the try?

Thank you for any input.

**Comment by Hedda** [comment]
See existing discussion -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478
